spanner/google/cloud/spanner/database.py

```diff
@@ -154,8 +158,13 @@ def ddl_statements(self):
     def spanner_api(self):
         """Helper for session-related API calls."""
         if self._spanner_api is None:
+            base_creds = self._instance._client.credentials
+            scoped = base_creds.with_scopes((SPANNER_DATA_SCOPE,))
```

Comment:
you need to check if these are scoped credentials. It's possible to use spanner with JWT credentials which are unscoped:

```python
if instance(base_creds, google.auth.credentials.Scoped):
    scoped = base_creds.with_scopes((SPANNER_DATA_SCOPE,))
```